tests/aws/services/events/test_events.py

```diff
@@ -996,7 +1088,323 @@ def _get_sqs_messages():
+        snapshot.add_transformer(snapshot.transform.regex(event_bus_name, "<bus-name>"))
+
+        rule_name_prefix = f"test-rule-{short_uid()}"
+        snapshot.add_transformer(snapshot.transform.regex(rule_name_prefix, "<rule-name-prefix>"))
+        count = 6
+
+        for i in range(count):
+            rule_name = f"{rule_name_prefix}-{i}"
+            put_rule(
+                Name=rule_name,
+                EventBusName=event_bus_name,
+                EventPattern=json.dumps(TEST_EVENT_PATTERN),
+            )
+
+        response = aws_client.events.list_rules(Limit=int(count / 2), EventBusName=event_bus_name)
+        snapshot.match("list-rules-limit", response)
+
+        response = aws_client.events.list_rules(
+            Limit=int(count / 2) + 2, NextToken=response["NextToken"], EventBusName=event_bus_name
+        )
+        snapshot.match("list-rules-limit-next-token", response)
+
+    @markers.aws.validated
+    @pytest.mark.skipif(
+        not is_v2_provider() and not is_aws_cloud(),
+        reason="V1 provider does not support this feature",
+    )
+    def test_describe_nonexistent_rule(self, aws_client, snapshot):
+        rule_name = f"this-rule-does-not-exist-1234567890-{short_uid()}"
+        snapshot.add_transformer(snapshot.transform.regex(rule_name, "<rule-name>"))
+
+        with pytest.raises(aws_client.events.exceptions.ResourceNotFoundException) as e:
+            aws_client.events.describe_rule(Name=rule_name)
+        snapshot.match("describe-not-existing-rule-error", e)
+
+    @markers.aws.validated
+    @pytest.mark.parametrize("bus_name", ["custom", "default"])
+    def test_disable_re_enable_rule(
+        self, create_event_bus, put_rule, aws_client, snapshot, bus_name
+    ):
+        if bus_name == "custom":
+            bus_name = f"bus-{short_uid()}"
+            snapshot.add_transformer(snapshot.transform.regex(bus_name, "<bus-name>"))
+            create_event_bus(Name=bus_name)
+
+        rule_name = f"test-rule-{short_uid()}"
+        snapshot.add_transformer(snapshot.transform.regex(rule_name, "<rule-name>"))
+        put_rule(
+            Name=rule_name,
+            EventPattern=json.dumps(TEST_EVENT_PATTERN),
+            EventBusName=bus_name,
+        )
+
+        response = aws_client.events.disable_rule(Name=rule_name, EventBusName=bus_name)
+        snapshot.match("disable-rule", response)
+
+        response = aws_client.events.describe_rule(Name=rule_name, EventBusName=bus_name)
+        snapshot.match("describe-rule-disabled", response)
+
+        response = aws_client.events.enable_rule(Name=rule_name, EventBusName=bus_name)
+        snapshot.match("enable-rule", response)
+
+        response = aws_client.events.describe_rule(Name=rule_name, EventBusName=bus_name)
+        snapshot.match("describe-rule-enabled", response)
+
+    @markers.aws.validated
+    def test_delete_rule_with_targets(
+        self, put_rule, sqs_create_queue, sqs_get_queue_arn, aws_client, snapshot
```

Comment:
good to know: the `put_rule` factory cleanup fails if the test case deletes the rule and pollutes the CI logs. More of a comment because we're doing quite badly everywhere with this.